test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -1306,3 +1307,32 @@ def test_diagram(h):
 
     assert h.diagram(mock_diagram) == expected
     mock_diagram.build_diagram.assert_called_with(h)
+
+
+def test_decompose():
+    ghz_circ1 = Circuit().ghz([0, 1, 2]).decompose()
+    ghz_circ2 = Circuit().ghz([0, 1, 2, 3]).decompose()
+    qft_circ = Circuit().qft([0, 1, 2]).decompose()
+    ghz_instr1 = Instruction(CompositeOperator.GHZ(3), [0, 1, 2])
+
+    assert ghz_circ1 == Circuit(ghz_instr1.decompose())
+    assert ghz_circ1 != ghz_circ2
+    assert ghz_circ1 != qft_circ
+
+
+def test_original_circuit_unchanged_after_decompose():
```

Comment:
I'm confused on what's being tested here. The name suggests that we test that the original circuit remains unchanged. However, we don't appear to be asserting that.